internal/controller/status/status_setters.go

```diff
@@ -430,6 +430,67 @@ func snippetsStatusEqual(status1, status2 ngfAPI.ControllerStatus) bool {
 	return ConditionsEqual(status1.Conditions, status2.Conditions)
 }
 
+func newAuthenticationFilterStatusSetter(status ngfAPI.AuthenticationFilterStatus, gatewayCtlrName string) Setter {
```

Comment:
Use a different name than `status` in the arg here, because you reuse it in the function and are overwriting it.